### PR TITLE
Add inventory section and SKU blocks

### LINKED ISSUE #37399: Create product SKU block and create inventory block
**Description**

This issue pertains to moving the SKU field over to the block based product editor.
This may warrant the creation of a new product sku block. I had a quick look and couldn't find a similar already existing block, but it would be worth having a re-look.

The product SKU block should work the same as it currently does within the product editor.

**Design**

Image

AkVNGImLgSqCObTQ3idVn7-fi-331-61736

**Acceptance criteria**

- [ ] Inventory section should be present within Inventory tab
- [ ] SKU field should be present within inventory section
- [ ] SKU should be auto generated from the product title, similar to how it works currently.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adding inventory section block along with SKU block and related logic.

Closes #37399 .

### How to test the changes in this Pull Request:

1. Enable the `block-editor-feature-enabled` feature flag. 
2. Browse to Products -> Add new
3. In the block editor, add a name in the name field and then click on the "Inventory tab"
4. You should see the Inventory section block with description and the SKU field
5. The SKU field should already be populated with a slug derived from the name that you added
6. Change the slug, and then save the product
7. Refresh the page to ensure the changes persist.
8. The name field should remain the same when you change the sku

### CODE REVIEW

File: packages/js/product-editor/src/components/details-name-block/edit.tsx [hunk 1]
```diff
@@ -76,10 +76,13 @@ export function Edit() {
 		}
 	);
 
-	const nameIsValid = useValidation(
-		'product/name',
-		() => Boolean( name ) && name !== AUTO_DRAFT_NAME
-	);
+	const nameIsValid = Boolean( name ) && name !== AUTO_DRAFT_NAME;
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/details-name-block/edit.tsx]: I think we should keep the validation hook. The validation hook prevent the editor from been saved. Also if it's removed then will be hard to identify/fix the logic once the validation updates will be implemented.

File: packages/js/product-editor/src/components/inventory-sku-block/block.json
Comment (mdperez86) [inline packages/js/product-editor/src/components/inventory-sku-block/block.json:1]: After this [comment](https://github.com/woocommerce/woocommerce/pull/37567#discussion_r1157906557) we started to create all new blocks inside the /blocks folder and also compacted the name of the block folder to something like `inventory-sku`.

**Review by mdperez86 — CHANGES_REQUESTED** [review]
Good job @joelclimbsthings !!! Just left some comments.

File: packages/js/product-editor/src/components/details-name-block/edit.tsx (hunk 1 above)
Comment (joelclimbsthings) [reply packages/js/product-editor/src/components/details-name-block/edit.tsx]: Good call 👍🏻 . Done in https://github.com/woocommerce/woocommerce/pull/37623/commits/56a187ad22483b602ff238c869e6982abcb4afaf

File: packages/js/product-editor/src/components/inventory-sku-block/block.json
Comment (joelclimbsthings) [reply packages/js/product-editor/src/components/inventory-sku-block/block.json:1]: I like it! That had been bothering me as well. Done in https://github.com/woocommerce/woocommerce/pull/37623/commits/9c6838745f4ae93fbe1d0460451a5686803902a3

**Comment by joelclimbsthings** [comment]
Hey @mdperez86 , thanks for the review! This has just been rebased onto trunk, and your changes were implemented. Let me know what you think.

File: packages/js/product-editor/src/blocks/inventory-sku/block.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "woocommerce/product-sku",
+	"title": "Product text control",
+	"category": "woocommerce",
+	"description": "The product sku.",
+	"keywords": [ "products", "sku" ],
+	"textdomain": "default",
+	"attributes": {
+		"name": {
```
Comment (mdperez86) [inline packages/js/product-editor/src/blocks/inventory-sku/block.json:11]: Just following @joshuatf's suggestions, see -> https://github.com/woocommerce/woocommerce/pull/37683#discussion_r1164497151

**Review by mdperez86 — APPROVED** [review]
Just forwarding one comment from @joshuatf. Not mandatory but something we could consider.

File: packages/js/product-editor/src/blocks/inventory-sku/block.json (hunk 2 above)
Comment (joelclimbsthings) [reply packages/js/product-editor/src/blocks/inventory-sku/block.json:11]: Thanks @mdperez86 , I just [responded in that thread](https://github.com/woocommerce/woocommerce/pull/37683#discussion_r1166040083).

Comment (joelclimbsthings) [reply packages/js/product-editor/src/blocks/inventory-sku/block.json:11]: Thanks @mdperez86 , I just [responded in that thread](https://github.com/woocommerce/woocommerce/pull/37683#discussion_r1166045984).

**Pull request merged**